AppController/lib/app_controller_client.rb

```diff
@@ -108,39 +115,34 @@ def make_call(time, retry_on_except, callr)
     refused_count = 0
     max = 5
 
+    # Do we need to retry at all?
+    if not retry_on_except
+      refused_count = max + 1
+    end
+
     begin
       Timeout::timeout(time) {
         yield if block_given?
       }
-    rescue Errno::ECONNREFUSED, Errno::EHOSTUNREACH
+    rescue Timeout::Error
+      Djinn.log_warn("[#{callr}] SOAP call to #{@ip} timed out")
+      raise FailedNodeException.new("Time out: is the AppController running?")
+    rescue Errno::ECONNREFUSED, Errno::EHOSTUNREACH,
+      OpenSSL::SSL::SSLError, NotImplementedError, Errno::EPIPE,
+      Errno::ECONNRESET, SOAP::EmptyResponseError, Exception => e
+      trace = e.backtrace.join("\n")
+      Djinn.log_warn("[#{callr}] exception in make_call to #{@ip}: #{e.class}\n#{trace}")
       if refused_count > max
-        raise FailedNodeException.new("Connection was refused. Is the " +
-          "AppController running?")
+        raise FailedNodeException.new("[#{callr}] failed to interact with #{@ip}.")
       else
         refused_count += 1
-        Kernel.sleep(1)
+        Kernel.sleep(3)
```

Comment:
The rationale is that, in the previous implementation we were re-trying forever. In this version (ie same implementation for all similar call) we need to cap-them to a maximum time, so I wanted to give it a bit more more time to recover (ie 15 seconds instead of 5). Perhaps we should just rely on the timeout as passed in the function (ie once the total time spent in retrying here is past timeout we fail)?